src/main/java/rife/scheduler/Task.java

```diff
@@ -421,16 +421,12 @@ public boolean equals(Object object) {
 
         var other_task = (Task) object;
 
-        if (other_task.getId() == getId() &&
+        return other_task.getId() == getId() &&
```

Comment:
This was purposefully written as separate statements to reduce the number of conditions to parse

Reply:
That doesn't make any sense to me. It's not saving anything. It still has to evaluate the `if` before the return. This is the decompiled code from the original class:

```java
 return other_task.getId() == this.getId() && other_task.getType().equals(this.getType()) && other_task.getPlanned() == this.getPlanned() && (null == other_task.getFrequency() && null == this.getFrequency() || other_task.getFrequency() != null && other_task.getFrequency().equals(this.getFrequency())) && other_task.getTaskManager() == this.getTaskManager();
```

which is exactly what I did.

Reply:
Not to parse by the compiler, to parse by the human reader